src/components/editor_keymap_printer.rs

```diff
@@ -0,0 +1,236 @@
+            ),
+        }
+    }
+
+    pub fn has_content(&self) -> bool {
+        self.key_meanings
+            .iter()
+            .any(|v| v.iter().any(Option::is_some))
+    }
+
+    fn keyboard_layout_to_keymaps(
+        keyboard_layout: &KeyboardLayout,
+        modifiers: KeyModifiers,
+        keymaps: &Vec<Keymaps>,
+    ) -> Vec<Vec<Option<String>>> {
+        keyboard_layout
+            .iter()
+            .map(|row| {
+                row.iter()
+                    .map(|key| {
+                        let ke = match modifiers {
+                            KeyModifiers::Shift => KeyEvent {
+                                code: KeyCode::Char(shifted(key).chars().next().unwrap()),
+                                modifiers: KeymapPrintSection::shifted_modifier(key),
+                            },
+                            _ => KeyEvent {
+                                code: KeyCode::Char(key.chars().next().unwrap()),
+                                modifiers: modifiers.clone(),
+                            },
+                        };
+
+                        match keymaps.iter().find_map(|p| p.get(&ke)) {
+                            // One may wonder why not simply return the short_description as it is
+                            // an Option<String> but by detecting None here and replacing it with
+                            // ??? we are able to see where we may not be providing a short
+                            // description in code.
+                            Some(keymap) => match keymap.short_description.clone() {
+                                Some(short) => Some(short),
+                                None => Some("???".to_string()),
+                            },
+                            // There truly is no key mapping for this key event.
+                            None => None,
+                        }
+                    })
+                    .collect()
+            })
+            .collect()
+    }
+
+    fn shifted_modifier(key: &'static str) -> KeyModifiers {
+        match key {
+            "," => KeyModifiers::None,
+            "." => KeyModifiers::None,
+            "/" => KeyModifiers::None,
+            ";" => KeyModifiers::None,
+            "'" => KeyModifiers::None,
+            _ => KeyModifiers::Shift,
+        }
+    }
+}
+
+type KeymapPrintSections = Vec<KeymapPrintSection>;
+
+fn collect_keymap_print_sections(layout: &KeyboardLayout) -> KeymapPrintSections {
+    let context = Context::default();
+    let mut editor = Editor::from_text(None, "");
+    let normal_keymaps: Vec<Keymaps> = editor.normal_mode_keymap_legend_config(&context).into();
+
+    editor.mode = Mode::MultiCursor;
+    let multicursor_keymaps: Vec<Keymaps> =
+        editor.normal_mode_keymap_legend_config(&context).into();
+
+    let mut vmode_keymaps = normal_keymaps.clone();
+    vmode_keymaps.insert(0, editor.visual_mode_initialized_keymaps());
+
+    let sections = vec![
```

Comment:
Use `[].to_vec()` instead of `vec![]` so that Syntax-based movement works for each element of this vector